### PR TITLE
fix: add 'Dashboard' to nav stack when resetting

### PR DESCRIPTION
**Pull request by snack-0verflow**
# Description

Added `Dashboard` to nav stack while resetting, also set current nav index to `SelectProfile`

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

_Please attach screenshots showing the change if relevant_

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevent
- [x] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

File: src/features/thank-you/ThankYouUKScreen.tsx [hunk 1]
```diff
@@ -132,8 +132,8 @@ export default class ThankYouUKScreen extends Component<RenderProps, State> {
                 <ClickableText
                   onPress={() =>
                     this.props.navigation.reset({
-                      index: 0,
-                      routes: [{ name: 'SelectProfile' }],
+                      index: 1,
+                      routes: [{ name: 'Dashboard' }, { name: 'SelectProfile' }],
```
Comment (abubuwe) [inline src/features/thank-you/ThankYouUKScreen.tsx:136]: Not 100% on how the screen naming works, but does this cover new + old homescreens (is the old one still called WelcomeRepeat)

Comment (snack-0verflow) [reply src/features/thank-you/ThankYouUKScreen.tsx:136]: True. Will update this

Comment (snack-0verflow) [reply src/features/thank-you/ThankYouUKScreen.tsx:136]: @abubuwe Used `appCoordinator.homeScreenName` to keep it dynamic

Comment (abubuwe) [reply src/features/thank-you/ThankYouUKScreen.tsx]: Can you confirm if this is needed to be done anywhere else with a quick text search?

Comment (snack-0verflow) [reply src/features/thank-you/ThankYouUKScreen.tsx]: Did a global search of `reset(` I believe this is the only place that requires a default home screen when resetting the stack. 

Other reset stack method does include a base route & or user is logged out.

**Pull request merged**